ds3-autogen-c/src/main/resources/templates/source_pre_init_functions.ftl

```diff
@@ -0,0 +1,502 @@
+}
+
+void ds3_client_proxy(ds3_client* client, const char* proxy) {
+    client->proxy = ds3_str_init(proxy);
+}
+
+void ds3_request_set_prefix(ds3_request* _request, const char* prefix) {
+    _set_query_param(_request, "prefix", prefix);
+}
+
+void ds3_request_set_metadata(ds3_request* _request, const char* name, const char* value) {
+
+    char* prefixed_name = g_strconcat("x-amz-meta-", name, NULL);
+
+    _set_header(_request, prefixed_name, value);
+
+    g_free(prefixed_name);
+}
+
+void ds3_request_reset_byte_range(ds3_request* _request) {
+    g_hash_table_remove(_request->headers, "Range");
+}
+
+void ds3_request_set_byte_range(ds3_request* _request, int64_t rangeStart, int64_t rangeEnd) {
+    char* range_value;
+
+    gpointer header_value = g_hash_table_lookup(_request->headers, "Range");
+    if (header_value != NULL) {
+        range_value = g_strdup_printf("%s,%ld-%ld", (char*)header_value, rangeStart, rangeEnd);
+    } else {
+        range_value = g_strdup_printf("bytes=%ld-%ld", rangeStart, rangeEnd);
+    }
+
+    _set_header(_request, "Range", range_value);
+    g_free(range_value);
+}
+
+void ds3_request_set_custom_header(ds3_request* _request, const char* header_name, const char* header_value) {
+   _set_header(_request, header_name, header_value);
+}
+
+void ds3_request_set_bucket_name(ds3_request* _request, const char* bucket_name) {
+    _set_query_param(_request, "bucket_id", bucket_name);
+}
+
+void ds3_request_set_creation_date(ds3_request* _request, const char* creation_date) {
+    _set_query_param(_request, "creation_date", creation_date);
+}
+
+void ds3_request_set_md5(ds3_request* _request, const char* md5) {
+    struct _ds3_request* request = (struct _ds3_request*) _request;
+    request->checksum_type = DS3_MD5;
+    request->checksum = ds3_str_init(md5);
+}
+
+void ds3_request_set_sha256(ds3_request* _request, const char* sha256) {
+    struct _ds3_request* request = (struct _ds3_request*) _request;
+    request->checksum_type = DS3_SHA256;
+    request->checksum = ds3_str_init(sha256);
+}
+
+void ds3_request_set_sha512(ds3_request* _request, const char* sha512) {
+    struct _ds3_request* request = (struct _ds3_request*) _request;
+    request->checksum_type = DS3_SHA512;
+    request->checksum = ds3_str_init(sha512);
+}
+
+void ds3_request_set_crc32(ds3_request* _request, const char* crc32) {
+    struct _ds3_request* request = (struct _ds3_request*) _request;
+    request->checksum_type = DS3_CRC32;
+    request->checksum = ds3_str_init(crc32);
+}
+
+void ds3_request_set_crc32c(ds3_request* _request, const char* crc32c) {
+    struct _ds3_request* request = (struct _ds3_request*) _request;
+    request->checksum_type = DS3_CRC32C;
+    request->checksum = ds3_str_init(crc32c);
+}
+
+void ds3_request_set_delimiter(ds3_request* _request, const char* delimiter) {
+    _set_query_param(_request, "delimiter", delimiter);
+}
+
+void ds3_request_set_marker(ds3_request* _request, const char* marker) {
```

Comment:
This is modifying a (bulk) request, and isn't an actual API call - nothing in the contract, more of a helper.